src/NLog/Targets/Target.cs

```diff
@@ -337,7 +337,7 @@ public void WriteAsyncLogEvents(params AsyncLogEventInfo[] logEvents)
         /// Writes the array of log events.
         /// </summary>
         /// <param name="logEvents">The log events.</param>
-        internal void WriteAsyncLogEvents(IList<AsyncLogEventInfo> logEvents)
+        public void WriteAsyncLogEvents(IList<AsyncLogEventInfo> logEvents)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3776)